src/hpguppi_ata_blade_obs_rawdisk_thread.c

```diff
@@ -358,25 +367,37 @@ static void *run(hashpipe_thread_args_t * args)
         if(directio) {
           open_flags |= O_DIRECT;
         }
-        for(i = 0; i < nbeams; i++){
-          sprintf(fname, "%s-beam%04d.%04d.raw", pf.basefilename, i, filenum);
 
+        for(i = 0; i < nfdraws; i++){
+          if(nbeams==0){
+            sprintf(fname, "%s.raw", pf.basefilename);
+          }
+          else{
+            sprintf(fname, "%s-beam%04d.%04d.raw", pf.basefilename, i, filenum);
+          }
+          
           fprintf(stderr, "Opening first raw file '%s' (directio=%d)\n", fname, directio);
           fdraws[i] = open(fname, open_flags, 0644);
           if (fdraws[i]==-1) {
             hashpipe_error(thread_name, "Error opening file.");
             pthread_exit(NULL);
           }
         }
-
+        
       }
 
       /* See if we need to open next file */
       if (block_count >= blocks_per_file) {
         filenum++;
-        for(i = 0; i < nbeams; i++){
+        
+        for(i = 0; i < nfdraws; i++){
           close(fdraws[i]);
-          sprintf(fname, "%s-beam%04d.%04d.raw", pf.basefilename, i, filenum);
+          if(nbeams==0){
+            sprintf(fname, "%s.raw", pf.basefilename);
```

Comment:
`sprintf(fname, "%s.%04d.raw", pf.basefilename, filenum);` please

otherwise it's a redundant statement, and the second file overwrites the first, and this is the conventional solution